### PR TITLE
added candy-icons icon package

### PR DESCRIPTION
**Pull request by eneoli**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/candy-icons/default.nix
Comment (h7x4) [inline pkgs/data/icons/candy-icons/default.nix:15]: Let's use `fetchzip` to fix this stuff for us automatically. Also, SRI hashes using the `hash` attribute is the norm these days.

```suggestion
  src = fetchzip {
    url = "https://github.com/EliverLara/candy-icons/archive/37b4566b4bfcdad99131a236ef1fc74db89a302e.zip";
    hash = "sha256-fgey2MduS9alUKiFhZr6KTKhYa68zh6LzEIcg/amrOo=";
  };
```

File: pkgs/data/icons/candy-icons/default.nix
Comment (h7x4) [inline pkgs/data/icons/candy-icons/default.nix:3]: `pkgs` in the arguments makes this a lot harder to override, and I believe it might have other downsides as well. Let's expand this argument list.

Also, there doesn't seem to be any need for a C compiler toolchain, so we can use `stdenvNoCC`.

See my comment regarding `fetchzip`

```suggestion
{ stdenvNoCC
, fetchzip
, gtk3
}:

stdenvNoCC.mkDerivation {
```

File: pkgs/data/icons/candy-icons/default.nix
Comment (h7x4) [inline pkgs/data/icons/candy-icons/default.nix:25]: Since we're now using `fetchzip`, there's no need for unpacking. Also, every package should have `runHooks`s.

```suggestion
  installPhase = ''
    runHook preInstall
  
    mkdir -p $out/share/icons
    mv candy-icons-* $out/share/icons/candy-icons

    for theme in $out/share/icons/*; do
      gtk-update-icon-cache --force $theme
    done
    
    runHook postInstall
  '';
```

File: pkgs/data/icons/candy-icons/default.nix
Comment (h7x4) [inline pkgs/data/icons/candy-icons/default.nix:8]:
```suggestion
  nativeBuildInputs = [ gtk3 ];
```

File: pkgs/data/icons/candy-icons/default.nix
Comment (h7x4) [inline pkgs/data/icons/candy-icons/default.nix:26]: This is going to need a `meta` block: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes.

I think you should also create a commit that adds yourself to the maintainer list, so you can be marked as package maintainer. Have a look at [some of the other closed first-time contribution prs](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+label%3A%2212.+first-time+contribution%22+is%3Aclosed+init) to see how this might look like.

**Review by h7x4 — CHANGES_REQUESTED** [review]
Hello @eneoli, welcome to nixpkgs!

Here's a few things you'll need to fix up before this PR can get merged.

File: pkgs/data/icons/candy-icons/default.nix
Comment (h7x4) [inline pkgs/data/icons/candy-icons/default.nix:35]: Seems to be a missing `}` here? Also, let's set platforms as well!

```suggestion
    maintainers = with maintainers; [ eneoli ];
    platforms = platforms.all;
  };
```

File: pkgs/data/icons/candy-icons/default.nix
Comment (h7x4) [inline pkgs/data/icons/candy-icons/default.nix:7]: nit:
```suggestion
```

**Review by h7x4 — CHANGES_REQUESTED** [review]
You'll have to rebase the commits to conform to the [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions). It should look something like this:

```
maintainers: add eneoli
candy-icons: init at 0.6.8
```

**Comment by MikaelFangel** [comment]
@eneoli - The commit adding you to the maintainers list should appear before the commit adding the icon package, and the package should be placed under `pkgs/by-name/so/some-example/package.nix` hierarchy if possible.

**Comment by eneoli** [comment]
@MikaelFangel huh? Many icon packages are under pkgs/data/icons. What’s the difference? That seemed more reasonable for me.

**Comment by h7x4** [comment]
@eneoli We're currently in the process of moving towards a new system, [`pkgs/by-name`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md). While I haven't seen anywhere that states it as a hard requirement for new packages, the package will probably be moved at some point soon anyways.

File: pkgs/data/icons/candy-icons/default.nix
Comment (eclairevoyant) [inline pkgs/data/icons/candy-icons/default.nix:33]: gpl3 is a deprecated identifier, please use gpl3Only or gpl3Plus as per upstream

File: pkgs/data/icons/candy-icons/default.nix
Comment (eclairevoyant) [inline pkgs/data/icons/candy-icons/default.nix:13]: Why is this pulling a random commit? Also can we use `fetchFromGitHub` here?

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

**Comment by MikaelFangel** [comment]
It's correct that it is not mandatory as some package currently have issues under the 'by-name' hierarchy but when that is said it is per [this readme](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md) written as the preferred way of adding new packages. 😊

File: pkgs/data/icons/candy-icons/default.nix
Comment (eneoli) [reply pkgs/data/icons/candy-icons/default.nix:13]: Unfortunately the maintainer of the package does not have some sort of release system on github nor tags for the versions.

Comment (eneoli) [reply pkgs/data/icons/candy-icons/default.nix]: FYI: https://github.com/EliverLara/candy-icons/issues/59

Comment (h7x4) [reply pkgs/data/icons/candy-icons/default.nix]: I'm not entirely sure about this, but I think it would okay to mark it with an unstable version since we're fetching from github. The version number only seems to be mentioned on https://www.opendesktop.org/p/1305251/

Comment (eneoli) [reply pkgs/data/icons/candy-icons/default.nix]: It's been 3 weeks since I asked the maintainer of candy-icons for some sort of version system. Sadly didn't receive a response. So how should we proceed? Just mark it as unstable?

Comment (h7x4) [reply pkgs/data/icons/candy-icons/default.nix]: Yes, I think it should be marked as unstable.

**Comment by D3vil0p3r** [comment]
Duplicated https://github.com/NixOS/nixpkgs/pull/259294

**Pull request closed without merging**
